images/fcos-bastion-image/root/usr/bin/clean-registries.sh

```diff
@@ -1,11 +1,25 @@
 #!/bin/sh
 set -x
 
+wait_for_container_running() {
+  local container_name=$1
+  echo "Waiting for container $container_name to become running..."
+  while ! podman inspect -f '{{.State.Running}}' "$container_name" >/dev/null 2>&1; do
+      sleep 1
+  done
```

Comment:
Can we add a timeout or do something like:

```
for i in {1..60} MAX; do
  if [ $i == MAX ]; then
    return 1
  fi
  if [ $(podman inspect -f '{{.State.Health.Status}}' "$container_name" >/dev/null 2>&1) == "healthy" ]; then
    echo "Container $container_name is now running"
    return 0
  fi
done
echo "The container didn't get ready in time"
return 1
```

Reply:
So State.Health.Status is not really working:

core@dhcp-1-235-109:~$ sudo podman inspect -f '{{.State.Health.Status}}' registry-5000

core@dhcp-1-235-109:~$ sudo podman inspect -f '{{.State.Running}}' registry-5000
true

Reply:
@aleskandro  health status is empty though container is running:
HealthcheckOnFailureAction": "none",
Error: no such object: "{{.State.Health.Status}}"
core@dhcp-1-235-109:~$ sudo podman inspect  '{{.State.Health.Status}}' registry-5000 |grep -i health -A3
Error: no such object: "{{.State.Health.Status}}"               "Health": {
                    "Status": "",
                    "FailingStreak": 0,
                    "Log": null
--
               "HealthcheckOnFailureAction": "none",
               "CreateCommand": [
                    "/usr/bin/podman",
                    "run",

core@dhcp-1-235-109:~$ sudo podman inspect  '{{.State.Health.Status}}' registry-5000 |grep -i running -A3
               "Status": "running",
               "Running": true,

Reply:
tested with Running status and timeout:
Waiting for container registry-6002 to become running...
+ for i in {1..60} MAX
+ '[' 1 == MAX ']'
+ podman inspect -f '{{.State.Running}}' registry-6002
+ sleep 1
+ for i in {1..60} MAX
+ '[' 2 == MAX ']'
+ podman inspect -f '{{.State.Running}}' registry-6002
+ echo 'Container registry-6002 is now running'
Container registry-6002 is now running
+ return 0
+ podman exec registry-6002 /bin/registry garbage-collect /etc/docker/registry/config.yml
2024-02-28 14:32:13.280454725 +0000 UTC m=+0.045342333 container exec ac32e58b16c2ba05914f0935a2d6d620cc66a6760548639ff751ab0def896f71 (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
0 blobs marked, 0 blobs and 0 manifests eligible for deletion
2024-02-28 14:32:13.293506963 +0000 UTC m=+0.058394571 container exec_died ac32e58b16c2ba05914f0935a2d6d620cc66a6760548639ff751ab0def896f71 (image=quay.io/libpod/registry:2.8.2, name=registry-6002, PODMAN_SYSTEMD_UNIT=registry@6002.service)
+ sed -i -e '/readonly/{n;s/enabled:.*/enabled: false/;}' /opt/registry-6002/config.yaml
+ systemctl restart registry@6002.service
+ exit 0